### PR TITLE
Add docstrings and comments

### PR DESCRIPTION
**Pull request by LuEdRaMo**

### CODE REVIEW

**Comment by lbenet** [comment]
Hi @LuEdRaMo! Thanks for opening this PR to improve the docstrings/comments of the code, and sorry for the long delay to react... I was on holidays 😄. In the next few days I'll review this PR and send you some feedback. If you still are in the development process, let me know it, and then I'll review when you think it is ready.

File: src/asteroid_dynamical_models.jl
Comment (lbenet) [inline src/asteroid_dynamical_models.jl:119]: Instead of "Apophis" I think it is better to write "asteroid" or "NEO"; the first suggestion is used in other files. Note that the same comment applies below and to other files.

File: src/process_radar_data_jpl.jl
Comment (lbenet) [inline src/process_radar_data_jpl.jl:90]: wheter -> whether
reciever -> receiver

File: src/propagation.jl
Comment (lbenet) [inline src/propagation.jl:74]: Should `differentiate` be used here, or the specialized method defined in PlanetaryEphemeris.jl? (cc: @PerezHz )

**Review by lbenet — COMMENTED** [review]
First, my apologies for the long time to review this PR... This PR is somewhat close to https://github.com/PerezHz/PlanetaryEphemeris.jl/pull/2 and thus some suggestions/comments apply here as well. This is part of the reason I only send few comments in the code.

The following are my general comments/suggestions:
- Move all `const`s to a `constants.jl` file: relevant for 'src/NEOs.jl',  'src/asteroid_dynamical_models.jl', and 'src/initial_conditions.jl'
- Write "asteroid" instead of "Apophis"; I think it is more general. This does not apply to src/delay_doppler.jl, since the files used there are specific to Apophis. TODO: Think on a "more general" interface, not specific for Apophis.
- Some functions in 'src/integration_methods.jl' include the name "apophis"; I think it is better to name them as "neo" or "asteroid", which matter for some notebooks and for the scripts. Perhaps we should open an issue with this proposal to make this change in another PR.

cc @PerezHz

File: src/process_radar_data_jpl.jl
Comment (LuEdRaMo) [reply src/process_radar_data_jpl.jl]: Done.

**Comment by LuEdRaMo** [comment]
- I've moved most constants to a new file `src/constants.jl`.
- I've changed **apophis** for **asteroid** mostly in `src/asteroid_dynamical_models.jl`. In my opinion all other occurrences are apophis-specific. 
- Right now NEOs can't be compiled, since it requires v0.10 of [TaylorSeries](https://github.com/JuliaDiff/TaylorSeries.jl) while [PlanetaryEphemeris.jl](https://github.com/PerezHz/PlanetaryEphemeris.jl) requires v0.11. Therefore in the last commit I suggest to change [TaylorSeries](https://github.com/JuliaDiff/TaylorSeries.jl) compatibility to include v0.11.

File: Project.toml
Comment (lbenet) [inline Project.toml:43]: What about of extending this to include TaylorSeries v0.12, which is the most recent one?

**Review by lbenet — APPROVED** [review]
I have left a single comment, asking whether we should consider increasing the possible versions of TaylorSeries to include v0.12 or not.

A last question: did you check that everything behaves identically *after* moving all the constants to a file? 

Aside from that, LGTM and I am in favor of merging... but I wait for your answer to the question above before doing that. Thanks a lot again!

File: src/propagation.jl
Comment (PerezHz) [reply src/propagation.jl:74]: Sorry for the late reply! I recall correctly, I avoided bumping TaylorSeries to the latest version in `NEOs.jl` precisely due to this issue, so probably it makes sense to use the other method (i.e., the `order`-preserving derivative), but we would have to check

**Comment by LuEdRaMo** [comment]
- I've run `pha/apophis.jl` and the results of the integration do not change with the commits in this PR.
- I've executed `add PlanetaryEphemeris` in `Pkg` to update `Manifest.toml`. Before this change, `NEOs` could not be compilated.

File: Project.toml
Comment (LuEdRaMo) [reply Project.toml:43]: Currently, we cannot update `TaylorSeries` to the latest version v0.12 since `PlanetaryEphemeris` requires at most v0.11. Therefore, we need to update first in `PlanetaryEphemeris` and then in `NEOs`.

File: src/propagation.jl
Comment (LuEdRaMo) [reply src/propagation.jl]: I've run `pha/apophis.jl` and the results do not change by using `PlanetaryEphemeris.ordpres_differentiate`.

**Comment by LuEdRaMo** [comment]
I've run `pha/apophis.jl` again and the results do not change by using `PlanetaryEphemeris.ordpres_differentiate`.

File: src/asteroid_dynamical_models.jl
Comment (LuEdRaMo) [reply src/asteroid_dynamical_models.jl]: I've changed apophis for asteroid mostly in `src/asteroid_dynamical_models.jl`. In my opinion all other occurrences are apophis-specific.

File: src/propagation.jl
Comment (lbenet) [reply src/propagation.jl]: Excellent!

File: Project.toml
Comment (lbenet) [reply Project.toml:43]: TaylorSeries v0.12.2 includes a fix on `promotion_rule` that solves some ambiguities, and a bug in `tan`; the latter is more important in the context of PlanetaryEphemeris. I did a quick search for `tan` and I think it only appears in the computation of the osculating terms. If this is correct, there shouldn't be any problem to bump the version of TaylorSeries used in PlanetaryEphemeris and here, and we should perhaps have it 0.12 only.

**Comment by lbenet** [comment]
> I've run pha/apophis.jl again and the results do not change by using PlanetaryEphemeris.ordpres_differentiate.

Excellent! I think we should decide about the minim version of TaylorSeries for both PlanetaryEphemeris and NEOs, and then merge the PRs, and bump to a *minor* version.

**Comment by LuEdRaMo** [comment]
After the last commit, neither
- updating `TaylorSeries` to v0.12,
- updating `PlanetaryEphemeris` to v0.2,
- using order-preserving differentiate

changes the results of `pha/apophis.jl`.

**Review by lbenet — APPROVED** [review]
Thanks a lot, @LuEdRaMo! LGTM!

I think we should bump a new (minor) version, and then merge!

**Comment by LuEdRaMo** [comment]
> Thanks a lot, @LuEdRaMo! LGTM!
> 
> I think we should bump a new (minor) version, and then merge!

Done.

**Comment by lbenet** [comment]
Thanks a lot once again @LuEdRaMo for all the work !

**Pull request merged**
